sovtokenfees/sovtokenfees/fees_authorizer.py

```diff
@@ -0,0 +1,58 @@
+from indy_common.authorize.authorizer import AbstractAuthorizer
+
+from indy_common.authorize.auth_constraints import AuthConstraint
+
+from indy_common.authorize.auth_actions import AbstractAuthAction
+from sovtokenfees.static_fee_req_handler import StaticFeesReqHandler
+from stp_core.common.log import getlogger
+
+from plenum.common.exceptions import UnauthorizedClientRequest
+
+
+logger = getlogger()
+
+FEES_FIELD_NAME = 'fees'
+
+
+class FeesAuthorizer(AbstractAuthorizer):
+    def __init__(self, fees_req_handler: StaticFeesReqHandler):
+        super().__init__()
+        self.fees_req_handler = fees_req_handler
+
+    def _get_fees_from_constraint(self, auth_constaint: AuthConstraint):
+        if auth_constaint.metadata:
+            if FEES_FIELD_NAME in auth_constaint.metadata:
+                return auth_constaint.metadata[FEES_FIELD_NAME]
+
+    def _can_pay_fees(self, request, required_fees):
+        try:
+            self.fees_req_handler.can_pay_fees(request, required_fees)
+        except Exception as e:
+            logger.warning("Encountered exception while can_pay_fees validation: {}".format(e))
+            return False
+        return True
+
+    def authorize(self,
```

Comment:
I think we need to check that the request sender's role matches the fees constraint.

Example: 
- 10 tokens for Trustee, and 20 for Steward.
- a txn is created with 10 tokens
- the txn is signed by both Trustee and Steward, but send by the Steward
Result: it should be rejected.